src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs

```diff
@@ -43,7 +43,8 @@ protected AbstractHealthCheck(TParameters parameters, ILogger logger, ActivitySo
 		/// <inheritdoc />
 		public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken token = default)
 		{
-			using Activity? activity = m_activitySource.StartActivity("HealthCheckActivity")
+			string activityName = string.IsNullOrWhiteSpace(context.Registration.Name) ? "HealthCheckActivity" : context.Registration.Name; 
```

Comment:
Can we differentiate between activities with identical names from different services? It's very likely that there will be collisions on registration name.

Reply:
Activity reports service name as one of the parameters, so it should be fine. https://github.com/microsoft/Omex/blob/4bc5d95363797390b29fb8e024b6952715f4cc7c/src/Extensions/Activities/Internal/EventSource/ActivityEventSender.cs#L84